jerry-main/fuzzilli.c

```diff
@@ -0,0 +1,371 @@
+      fzliout = stdout;
+    }
+
+    /* Based on the jerryx_handler_print handler */
+    const char *const null_str = "\\u0000";
+
+    for (jerry_length_t arg_index = 1; arg_index < args_cnt; arg_index++)
+    {
+      jerry_value_t str_val;
+
+      if (jerry_value_is_symbol (args_p[arg_index]))
+      {
+        str_val = jerry_get_symbol_descriptive_string (args_p[arg_index]);
+      }
+      else
+      {
+        str_val = jerry_value_to_string (args_p[arg_index]);
+      }
+
+      if (jerry_value_is_error (str_val))
+      {
+        /* There is no need to free the undefined value. */
+        ret_val = str_val;
+        break;
+      }
+
+      jerry_length_t length = jerry_get_utf8_string_length (str_val);
+      jerry_length_t substr_pos = 0;
+      jerry_char_t substr_buf[256];
+
+      do
+      {
+        jerry_size_t substr_size = jerry_substring_to_utf8_char_buffer(str_val,
+                                                                       substr_pos,
+                                                                       length,
+                                                                       substr_buf,
+                                                                       256 - 1);
+
+        jerry_char_t *buf_end_p = substr_buf + substr_size;
+
+        /* Update start position by the number of utf-8 characters. */
+        for (jerry_char_t *buf_p = substr_buf; buf_p < buf_end_p; buf_p++)
+        {
+          /* Skip intermediate utf-8 octets. */
+          if ((*buf_p & 0xc0) != 0x80)
+          {
+            substr_pos++;
+          }
+        }
+
+        if (substr_pos == length)
+        {
+          *buf_end_p++ = (arg_index < args_cnt - 1) ? ' ' : '\n';
+        }
+
+        for (jerry_char_t *buf_p = substr_buf; buf_p < buf_end_p; buf_p++)
+        {
+          char chr = (char)*buf_p;
+
+          if (chr != '\0')
+          {
+            putc(chr, fzliout);
+            continue;
+          }
+
+          for (jerry_size_t null_index = 0; null_str[null_index] != '\0'; null_index++)
+          {
+            putc(null_str[null_index], fzliout);
+          }
+        }
+      } while (substr_pos < length);
+
+      jerry_release_value(str_val);
+    }
+
+    if (args_cnt == 0 || jerry_value_is_error(ret_val))
+    {
+      putc('\n', fzliout);
+    }
+    fflush(fzliout);
+  }
+
+  jerry_heap_free(operation, str_size * sizeof(jerry_char_t));
+  return ret_val;
+}
+
+/**
+ * Register a JavaScript function in the global object.
+ */
+static void
+register_js_function(const char *name_p,                 /**< name of the function */
+                     jerry_external_handler_t handler_p) /**< function callback */
+{
+  jerry_value_t result_val = jerryx_handler_register_global((const jerry_char_t *)name_p, handler_p);
+
+  if (jerry_value_is_error(result_val))
+  {
+    jerry_port_log(JERRY_LOG_LEVEL_WARNING, "Warning: failed to register '%s' method.", name_p);
```

Comment:
It seems that this is the only place where `jerry_port_log` is used. Should this also be `fprintf`? Or should all other places be `jerry_port_log` calls? (Main message is: use one API and use it consistently.)

Reply:
I changed the code to use `fprintf(stderr,` everywhere.